core/model/collection.py

```diff
@@ -241,35 +271,17 @@ def by_protocol(cls, _db: Session, protocol: str | None) -> Query[Collection]:
 
         return qu
 
-    @classmethod
-    def by_datasource(
-        cls, _db: Session, data_source: DataSource | str
-    ) -> Query[Collection]:
-        """Query collections that are associated with the given DataSource.
-
-        Collections marked for deletion are not included.
-        """
-        data_source_name = (
-            data_source.name if isinstance(data_source, DataSource) else data_source
-        )
-
-        qu = (
-            _db.query(cls)
-            .join(
-                IntegrationConfiguration,
-                cls.integration_configuration_id == IntegrationConfiguration.id,
-            )
-            .filter(
-                IntegrationConfiguration.settings_dict[
-                    Collection.DATA_SOURCE_NAME_SETTING
-                ].astext
-                == data_source_name
-            )
-            .filter(Collection.marked_for_deletion == False)
-        )
-        return qu
+    @property
+    def name(self) -> str:
+        """What is the name of this collection?"""
+        if self.integration_configuration is None:
+            raise ValueError("Collection has no integration configuration.")
+        name = self.integration_configuration.name
+        if not name:
+            raise ValueError("Collection has no name.")
+        return name
 
-    @hybrid_property
+    @property
```

Comment:
This never made sense as a `hybrid_property` since this for loop would not work in the class method case, so I switch it to a `property`.